spec/immutable_struct_spec.rb

```diff
@@ -113,4 +113,57 @@ def lawsuit
       instance.to_h.should == {flappy: 'bird', lawsuit: 'pending'}
     end
   end
+
+  describe "equality" do
+
+    before do
+      klass_1 = ImmutableStruct.new(:foo, :bar)
+      klass_2 = ImmutableStruct.new(:foo, :bar)
+      @k1_a = klass_1.new(foo: 'foo', bar: 'bar')
+      @k1_b = klass_1.new(foo: 'xxx', bar: 'yyy')
+      @k1_c = klass_1.new(foo: 'foo', bar: 'bar')
+      @k2_a = klass_2.new(foo: 'foo', bar: 'bar')
+    end
+
+    describe "==" do
+
+      it "should be equal to itself" do
```

Comment:
@racurry I've rewritten to be explicit with return values.  Thanks for the call-out.